kitsune/questions/api.py

```diff
@@ -289,6 +302,29 @@ def take(self, request, pk=None):
 
         return Response(status=204)
 
+    @action(methods=['POST'], permission_classes=[permissions.IsAuthenticated])
+    def add_tag(self, request, pk=None):
+        question = self.get_object()
+        tag = request.DATA['tag']
+
+        try:
+            canonical_name = add_existing_tag(tag, question.tags)
+        except Tag.DoesNotExist:
+            if request.user.has_perm('taggit.add_tag'):
+                question.tags.add(tag)
+                canonical_name = tag
+            else:
+                raise GenericAPIException(403, 'You are not authorized to create new tags.')
+
+        return Response(canonical_name)
```

Comment:
It would probably be good to reuse the `QuestionTagSerializer` from above. Maybe something like

```
tag = Tag.objects.get(name=tag_name)
return Response(QuestionTagSerializer(instance=tag).data)
```